app/javascript/packs/leafletMap.js

```diff
@@ -64,14 +64,42 @@ export async function setupMap() {
   map.on("click", onClick);
 }
 
+function parsePosition(position){
+  return "?long="+position.lng+"&lat="+position.lat
+}
+
+function pinPopupContent(position) {
+  return "<ul>"+
+  "<li><a href=\"/rooms/new"+parsePosition(position)+"\">Add Room</a></li>"+
+  "<li><a href=\"/buildings/new"+parsePosition(position)+"\">Add Building</a></li>"+
+  "<li><a href=\"/locations/new"+parsePosition(position)+"\">Add Location</a></li>"+
+  "<li><a id=\"deletepin\" \" href=\"javascript:void(0);\" >Delete Pin</a></li>"+
+  "</ul>";
+}
+
 function addPin(e, pinNumber) {
   pins[pinNumber] = L.marker(e.latlng, { icon: pinIcons[pinNumber] });
   pins[pinNumber].addTo(map);
   pins[pinNumber].on("click", function (e) {
-    removePin(pinNumber);
+    pinOnClick(pinNumber, e.latlng);
```

Comment:
I think you shouldn't use `.on("click", ...)` here.
It might be better to use the Leaflet `bindPopup(...)` method ([https://leafletjs.com/reference.html#marker-bindpopup](https://leafletjs.com/reference.html#marker-bindpopup)) for this.